src/util/misc/projectStroke.ts

```diff
@@ -39,62 +49,216 @@ export const projectStrokeOnPoints = (
   openPath: boolean
-): Point[] => {
-  const coords: Point[] = [],
+): {projectedPoint: Point, originPoint: Point, bisector?: { vector: Point, angle: number }}[] => {
+
+  const coords: {projectedPoint: Point, originPoint: Point, bisector?: { vector: Point, angle: number }}[] = [],
     s = strokeWidth / 2,
+    scale = new Point(scaleX, scaleY),
     strokeUniformScalar = strokeUniform
       ? new Point(1 / scaleX, 1 / scaleY)
       : new Point(1, 1),
-    getStrokeHatVector = (v: Point) => {
-      const scalar = s / Math.hypot(v.x, v.y);
-      return new Point(
-        v.x * scalar * strokeUniformScalar.x,
-        v.y * scalar * strokeUniformScalar.y
-      );
+    scaleHatVector = (hatVector: Point, scalar: number) => {
+      return hatVector.multiply(strokeUniformScalar).scalarMultiply(scalar);
+    },
+    applySkew = (v: Point) => {
+      let vector = new Point(v);
+      vector.y += vector.x * Math.tan(degreesToRadians(skewY)) // skewY must be applied before skewX as this distortion affects skewX calculation
+      vector.x += vector.y * Math.tan(degreesToRadians(skewX))
+      return vector
     };
+
   if (points.length <= 1) {
     return coords;
   }
+
   points.forEach(function (p, index) {
-    const A = new Point(p.x, p.y);
-    let B, C;
+    let A = new Point(p.x, p.y), B, C;
     if (index === 0) {
       C = points[index + 1];
-      B = openPath
-        ? getStrokeHatVector(createVector(C, A)).add(A)
-        : points[points.length - 1];
-    } else if (index === points.length - 1) {
+      B = openPath ? A : points[points.length - 1];
+    }
+    else if (index === points.length - 1) {
       B = points[index - 1];
-      C = openPath ? getStrokeHatVector(createVector(B, A)).add(A) : points[0];
-    } else {
+      C = openPath ? A : points[0];
+    }
+    else {
       B = points[index - 1];
       C = points[index + 1];
     }
-    const bisector = getBisector(A, B, C),
+
+    //  safeguard in case `points` are not `Point`
+    B = new Point(B.x, B.y);
+    C = new Point(C.x, C.y);
+
+    /* OPEN PATH
+      If open path, no matter the type of the line join, the line is always the same
+      Calculation: to find the projections, just find the points orthogonal to that stroke 
+      Visually detailed here: https://github.com/fabricjs/fabric.js/issues/8025
+      TODO: take into account line-cap (I believe that should change the projections)
+    */
+    if (openPath && (index === 0 || index === points.length - 1)) {
+      var D = index === 0 ? C : B, 
+        // When the stroke is uniform, scaling affects the arrangement of points. So we must take it into account.
+        vector = createVector(strokeUniform ? A.multiply(scale) : A, strokeUniform ? D.multiply(scale) : D), 
+        hatOrthogonalVector = getOrthogonalUnitVector(vector), 
+        orthogonalVector = scaleHatVector(hatOrthogonalVector, s);
+
+      // We must bounce the projection to both sides, since we don't know which one is the outer edge
```

Comment:
This opens up the bug that we discussed, the case of a big `strokeWidth` value.
The wrong projection might cross over the shape's other side.
We should use intersection IMO.
`intersectLinePolygon` should do the job

Reply:
Then using the ray algorithm concept, `intersectedPoints % 2`

Reply:
I think that in this specific case there is no risk, because there is a stroke on both sides. 

The problem happened in `miter` that one side had a stroke and the other didn't (and this projection here that shits everything). But it was solved considering the direction of the bisector.

Reply:
I am off for a few days
When I vome back I'll review everything and see if I have a case or not.

Reply:
Okay, I'll update the documentation and then work on `skewX` and `skewY` applied at the same time.

Reply:
Take a look at codesandbox ( https://codesandbox.io/s/project-stroke-points-with-skew-correction-yz7c77?file=/src/index.js ). No matter the stroke size, both sides can always influence the BBOX calculation:
![openpath](https://user-images.githubusercontent.com/65685842/194463151-1c7aef21-7722-419a-b43a-67d936587115.gif)

The problem is in `miter`, when we project two sides, one of which has no stroke (observe the behavior of the red vector, if we took it into account, it would distort the BBOX calculation):

![openpath miter](https://user-images.githubusercontent.com/65685842/194463394-20c4c6b6-558b-4391-b412-84468e60290c.gif)

Reply:
I got confused.
I meant when `strokeLineCap === 'round'` but even in this case I don't see it is a possible case